### PR TITLE
Update contributing md

### PR DESCRIPTION
**Pull request by hansvancalster**
Maintenance PR to update the contributing guidelines and guidelines in pull request template.

### CODE REVIEW

**Comment by hansvancalster** [comment]
@ElsLommelen here also, feel free to improve and merge. No need to wait for my return to work.

**Review by ElsLommelen — APPROVED** [review]
I found some minor typo issues.  @florisvdh do you mind checking and merging?

File: CONTRIBUTING.md
Comment (ElsLommelen) [inline CONTRIBUTING.md:120]:
```suggestion
   These checks will run `protocolhelper::check_frontmatter()` and `protocolhelper::check_structure()`, and update the version number if needed.
```

File: CONTRIBUTING.md
Comment (ElsLommelen) [inline CONTRIBUTING.md:123]:
```suggestion
       1.  Address the problems detected by `protocolhelper::check_frontmatter` or `protocolhelper::check_structure`. You can see the list of problems by clicking on the check online (at the bottom of the pull request webpage). Alternatively, you can run these functions locally and see the list of problems printed in your console.
        Note that during this automatic run, a commit to update the version number may be automatically pushed to the remote.
        This should not pose a problem for local clones as this change can be automatically included and will not cause conflicts (i.e. `git pull origin main` with a local protocol branch checked out)
```

**Review by ElsLommelen — COMMENTED** [review]
While reviewing the release PR, I got inspiration to add this here as well.  ;-)

File: CONTRIBUTING.md
Comment (florisvdh) [inline CONTRIBUTING.md:36]:
```suggestion
In case you need another package than the ones installed (see the [DESCRIPTION file](DESCRIPTION)) for this project, proceed as follows:
```

File: CONTRIBUTING.md
Comment (florisvdh) [inline CONTRIBUTING.md:41]: This procedure may cause headache for novices. Perhaps provide a single `protocolhelper` function that accepts a vector of package names and does this all at once?

@hansvancalster @ElsLommelen this should not delay this PR, but if considered a useful idea, be turned into an issue at `protocolhelper`.

File: CONTRIBUTING.md
Comment (florisvdh) [reply CONTRIBUTING.md]: > I'm not sure confirming is the right word here.

@ElsLommelen I've edited your suggestion

File: CONTRIBUTING.md
Comment (florisvdh) [reply CONTRIBUTING.md:122]: This should not pose a problem for local clones as this change can be automatically included and will not cause conflicts (i.e. \`git pull origin main\` with a local protocol branch checked out)

@ElsLommelen I don't understand what you mean with the last sentence. Anyway it may be too scary for contributors and perhaps unneeded, so it can be dropped?

**Review by florisvdh — CHANGES_REQUESTED** [review]
Some minor things to be addressed; can you have a look @ElsLommelen and accept the commit suggestions if agreed?

File: CONTRIBUTING.md
Comment (ElsLommelen) [reply CONTRIBUTING.md:122]: I copied it more or less from a phrase of @hansvancalster in RELEASE.md, and it refers to the fact that the commit that is automatically generated in the GHA, should be pulled locally.  But you are right, it should be written less complicated.  What about something like `If this commit is added, press pull in the RStudio Git Pane to add it to your local clone.` to replace the last sentence?

As git often gives an error and asks to pull remote changes first before pushing additional commits, I would in any case write something, as this error may scare starting Github users as well.  (Or they may just click the message away without noticing their commit is not pushed to the origin.)

File: .github/pull_request_template.md
Comment (ElsLommelen) [reply .github/pull_request_template.md:21]: To be honest, I never tried this before, so I checked it right now, and I see no option to convert a PR to draft before clicking 'create pull request'.  I suppose this option only appears after the PR is created?

File: .github/pull_request_template.md
Comment (ElsLommelen) [reply .github/pull_request_template.md:11]: Eh, I'm not sure what this looks like when adding a PR, I never used these templates.  If this note is visible as written here when opening a PR and not shown after submitting, what I suppose it would do, it is fine for me.

(Personally I favor the use of `fixes #x` and dislike `closes` or maintainers that close issues before the solution is merged to the main branch, certainly from the viewpoint of a package user that encounters a problem and finds a similar issue closed already.  But this is personal and out of question here.  ;-) )

File: .github/pull_request_template.md
Comment (ElsLommelen) [reply .github/pull_request_template.md:16]: He might just have switched to Visual mode, which causes the html to be added?  I'll accept (ignore my other comment), and @hansvancalster can address this afterwards.

File: CONTRIBUTING.md
Comment (ElsLommelen) [reply CONTRIBUTING.md:41]: I have a related issue that I need to address with @hansvancalster: for now the renv packages are installed in the docker images (all three) itself, which has the advantage that the docker containers have a full installation already (which makes the GHA quick), but the disadvantage that we need to build a new docker image after adding new packages (which might be necessary anyway if a package installation needs additional software to run on Ubuntu).  (And also: is this installation of all renv packages necessary for all three GHA, i.e. in which steps do we need to render the protocols?)

In the current situation, we have to take action anyway (rebuild the docker image), so maybe in this case we can just add this package to renv as well?  Then they just have to ask us to add new packages, which has the advantage that we can also 'consider carefully' ourselves (see text on next lines).

And to be honest, regarding the many difficulties I experienced to have these 'renv' packages installed in docker, and considering the fact that also less experienced R users may add protocols, I think it may be a good idea anyway to save novices from a headache add these packages ourselves.  As these are protocols, I expect mainly text with some illustrations or tables, so I suppose the list of packages is rather limited and will hardly grow after the first dozen of packages are added?  Or am I mistaken and could it be a large effort for us?

Anyway, to be added either as an issue, either as a discussion topic in the PR I submit to add the automation...

File: .github/pull_request_template.md
Comment (florisvdh) [reply .github/pull_request_template.md:21]: I've rechecked; it can be done in a dropdown:

afbeelding

Of course both approaches are valid, I just like this one more :yum:

File: .github/pull_request_template.md
Comment (florisvdh) [reply .github/pull_request_template.md]: Huh, let's pray it's not VME mode doing this, but perhaps that's the cause indeed :confounded:.

File: CONTRIBUTING.md
Comment (florisvdh) [reply CONTRIBUTING.md:41]: Sounds all very reasonable to me. I leave the further considerations to you and Hans.

File: CONTRIBUTING.md
Comment (florisvdh) [reply CONTRIBUTING.md:122]: Your suggestion is OK indeed!

Now, it seems I was confused because the check-functions, either run in GHA or locally (line 122), cause a new commit to be 'pushed to the remote' (quoting your line 123 above). If that happened from the local repo, then pulling makes no sense. Maybe it doesn't happen that way when run locally, but that's not entirely clear here.

File: .github/pull_request_template.md
Comment (ElsLommelen) [reply .github/pull_request_template.md]: Yes, VME does put R comments into a html chunk (= the initial situation).  Why @hansvancalster only removed the double backticks and not the whole chunk around the comment, is also unclear to me.  Maybe some styling utility did it?

File: CONTRIBUTING.md
Comment (ElsLommelen) [reply CONTRIBUTING.md:122]: No, this new commit is pushed from the GHA to the remote by also running `update_version_number()` additionally to the check-functions.  So a local run of the check-functions will not cause a new commit (running `update_version_number()` locally will, but no need anymore to bother contributors with that and mention it in the guidelines).

File: .github/pull_request_template.md
Comment (florisvdh) [reply .github/pull_request_template.md:21]: I suppose many people don't know this indeed. Anyway your suggestion to collapse them in one item that (just shortly) refers the 2 possibilities, is very reasonable; feel free to add that :+1:.

File: CONTRIBUTING.md
Comment (florisvdh) [reply CONTRIBUTING.md:122]: Thanks for clarifying. Your suggested change (quoted below) is already sufficient to prevent further confusion, so fine to add that change!

> If this commit is added, press pull in the RStudio Git Pane to add it to your local clone.

File: .github/pull_request_template.md
Comment (ElsLommelen) [inline .github/pull_request_template.md:21]:
```suggestion
-   [ ] Open the dropdown (triangle) near 'create pull request' and choose 'Create draft pull request'
```

File: CONTRIBUTING.md
Comment (ElsLommelen) [reply CONTRIBUTING.md:122]: edited above.

**Review by ElsLommelen — COMMENTED** [review]
I suppose that's it for now?  I will make sure the issue on renv will be discussed with @hansvancalster afterwards, and I will first have a new look at the PR on release to change phrases that we rephrased here.

**Review by florisvdh — APPROVED** [review]
Nice work :+1:.

**Pull request merged**
